core/com.b2international.snowowl.datastore/src/com/b2international/snowowl/core/repository/RepositoryBuilder.java

```diff
@@ -145,6 +145,20 @@ public Repository build(Environment env) {
 				configurer.getAdditionalMappings().forEach(mappings::putMapping);
 			});
 		
+		if (deletionPolicy instanceof CompositeComponentDeletionPolicy) {
+			
+			CompositeComponentDeletionPolicy compositePolicy = (CompositeComponentDeletionPolicy) deletionPolicy;
+			
+			repositoryConfigurers.forEach(configurer -> {
+				configurer.getComponentDeletionPolicies().forEach( (clazz, predicate) -> {
+					if (!compositePolicy.getDeletionPolicies().containsKey(clazz)) {
```

Comment:
`compositePolicy.getDeletionPolicies().putIfAbsent(clazz, predicate);`

It's also a bit unexpected (for me) to see a builder on `CompositeComponentDeletionPolicy`, suggesting that the result is immutable, then proceed to pull a writable `Map` out via the get method and inject extra entries. A factory method could be added to the builder that starts with an existing `CompositeComponentDeletionPolicy`.